notifications.bs

```diff
@@ -802,92 +801,109 @@ attribute's getter must return the <a>maximum number of actions</a> supported.
+<a for=notification>data</a>, <a>this</a>'s <a>relevant Realm</a>). If this throws an exception,
+then return null.
 <!-- Relies on SameObject becoming Cached as proposed -->
 
-<p>The <dfn attribute for=Notification><code>actions</code></dfn> attribute's getter must return the
-result of the following steps:
+<p>The <dfn attribute for=Notification><code>actions</code></dfn> getter steps are:
 
 <ol>
-  <li><p>Let <var>frozenActions</var> be an empty list of type {{NotificationAction}}.
+ <li><p>Let <var>frozenActions</var> be an empty list of type {{NotificationAction}}.
 
-  <li><p>For each <var>entry</var> in the <a>notification</a>'s list of
-  <a for=notification>actions</a>, perform the following steps:
+ <li>
+  <p><a for=list>For each</a> <var>entry</var> of <a>this</a>'s <a>notification</a>'s list of
+  <a for=notification>actions</a>:
 
   <ol>
-    <li><p>Let <var>action</var> be a new {{NotificationAction}}.
+   <li><p>Let <var>action</var> be a new {{NotificationAction}}.
 
-    <li><p>Set <var>action</var>'s {{NotificationAction/action}} to
-    <var>entry</var>'s <a for=action>name</a>.
+   <li><p>Set <var>action</var>'s {{NotificationAction/action}} to <var>entry</var>'s
+   <a for=action>name</a>.
 
-    <li><p>Set <var>action</var>'s {{NotificationAction/title}} to
-    <var>entry</var>'s <a for=action>title</a>.
+   <li><p>Set <var>action</var>'s {{NotificationAction/title}} to <var>entry</var>'s
+   <a for=action>title</a>.
 
-    <li><p>Set <var>action</var>'s {{NotificationAction/icon}} to
-    <var>entry</var>'s <a for=action>icon URL</a>.
+   <li><p>Set <var>action</var>'s {{NotificationAction/icon}} to <var>entry</var>'s
+   <a for=action>icon URL</a>.
 
-    <!-- XXX IDL dictionaries are usually returned by value, so don't need to be
-         immutable. But FrozenArray reifies the dictionaries to mutable JS
-         objects accessed by reference, so we explicitly freeze them.
-         It would be better to do this with IDL primitives instead of JS - see
-         https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
-    <li><p>Call <a lt="freeze">Object.freeze</a> on <var>action</var>, to
-    prevent accidental mutation by scripts.
+   <!-- XXX IDL dictionaries are usually returned by value, so don't need to be
+        immutable. But FrozenArray reifies the dictionaries to mutable JS
+        objects accessed by reference, so we explicitly freeze them.
+        It would be better to do this with IDL primitives instead of JS - see
+        https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
+   <li><p>Call <a lt="freeze">Object.freeze</a> on <var>action</var>, to
+   prevent accidental mutation by scripts.
 
-    <li><p>Append <var>action</var> to <var>frozenActions</var>.
+   <li><p>Append <var>action</var> to <var>frozenActions</var>.
   </ol>
 
-  <li><p><a>Create a frozen array</a> from <var>frozenActions</var>.
+ <li><p><a>Create a frozen array</a> from <var>frozenActions</var>.
```

Comment:
You forgot to return the result of "create a frozen array". (Preexisting problem.)